src/eradiate/scenes/surface/_dem.py

```diff
@@ -1,50 +1,320 @@
-      prevents object referencing.
+    mesh : :class:`.BufferMeshShape`
+        A triangulated mesh representing the DEM
+
+    theta_lim : tuple
+        The limits of the latitude extent of the DEM
+
+    phi_lim : tuple
+        The limits of the longitude extent of the DEM
+    """
+
+    coord_units = da.attrs["coordinate_units"]
+    if "lat" in da.coords and "lon" in da.coords:
+        lat = ureg.Quantity(da.lat.values, coord_units).m_as(ureg.deg)
+        lon = ureg.Quantity(da.lon.values, coord_units).m_as(ureg.deg)
+        vertices = _generate_dem_vertices(lat, lon, to_quantity(da))
+        faces = _generate_face_indices(len(lat), len(lon))
+
+    elif "x" in da.coords and "y" in da.coords:
+        if isinstance(geometry, SphericalShellGeometry):
+            planet_radius = geometry.planet_radius
+
+        x = np.rad2deg(ureg.Quantity(da.x.values, coord_units) / planet_radius).m
+        y = np.rad2deg(ureg.Quantity(da.y.values, coord_units) / planet_radius).m
+
+        vertices = _generate_dem_vertices(x, y, to_quantity(da))
+        faces = _generate_face_indices(len(x), len(y))
+
+    theta_lim, phi_lim = _minmax_coordinates(vertices)
+    theta_mean, phi_mean = _mean_coordinates(vertices)
+    atmo_bottom = geometry.ground_altitude
+
+    if isinstance(geometry, SphericalShellGeometry):
+        trafo = (
+            mi.ScalarTransform4f.rotate(axis=[0, 0, 1], angle=-90)
+            @ mi.ScalarTransform4f.rotate(axis=[0, 1, 0], angle=-(90 - theta_mean))
+            @ mi.ScalarTransform4f.rotate(axis=[0, 0, 1], angle=-phi_mean)
+        ).matrix.numpy()[:3, :3]
+
+        vertices_spherical = _transform_vertices_spherical_shell(
+            vertices, geometry.planet_radius
+        )
+        vertices_new = [trafo.dot(vertex) for vertex in vertices_spherical]
+
+        mesh = BufferMeshShape(vertices=vertices_new, faces=faces)
+    elif isinstance(geometry, PlaneParallelGeometry):
+        vertices_new = _transform_vertices_plane_parallel(
+            vertices,
+            planet_radius=EARTH_RADIUS,
+            altitude=atmo_bottom,
+        )
+        mesh = BufferMeshShape(vertices=vertices_new, faces=faces)
+
+    return mesh, theta_lim, phi_lim
+
+
+def _generate_dem_vertices(x, y, elevation):
+    """
+    Generate DEM vertex positions as (latitude, longitude, elevation) pairs
+    for a plane parallel atmosphere geometry.
+
+    Note
+    ----
+    Elevation data is converted into kernel units and the unit is removed.
+
+    Parameters
+    ----------
+    latitude_points : array-like
+        Points along the latitude axis of the DEM
+
+    longitude_points : array-like
+        Points along the longitude axis of the DEM
+
+    elevation : quantity
+        Elevation data in units of length.
+
+    Returns
+    -------
+    vertices : array
+        DEM vertices as (lat, lon, elevation) triplets. Atmospheric geometry
+        is not taken into account at this step
```

Comment:
The atmosphere has nothing to do with this part of the code, so let's not confuse people with it:
```suggestion
        DEM vertices as (lat, lon, elevation) triplets.
```